### DISCUSSION TITLE
Fitur pesan ke Grup telah tersedia! - Rilis v1.0-RC2

### DISCUSSION THREAD

**Discussion by rioastamal** [kirimwa-id/forum#19] [Pengumuman]
Halo semua,

Dengan senang hati saya mengumumkan jika salah satu fitur yang paling banyak di request di roadmap yaitu Kirim Pesan ke Grup #4 telah tersedia pada API KirimWA v1.0-RC2. 🥳 

### Cara Penggunaan

Untuk mengirim pesan ke grup tetap menggunakan endpoint `POST /messages`. Berikut adalah contoh mengirim pesan teks ke grup dengan ID `6281234567890-1628639680`. Referensi lengkap lihat [dokumentasi API](https://developer.kirimwa.id/api/v1/#messages).

```bash
curl -X POST 'https://api.kirimwa.id/v1/messages' \
  -H 'Authorization: Bearer API_TOKEN' \
  -H 'Content-Type: application/json' \
  -d '
{
  "phone_number": "6281234567890-1628639680",
  "message": "Halo semua ini dari api.kirimwa.id",
  "device_id": "iphone-x-pro",
  "message_type": "text",
  "is_group_message": true
}
'
```

Untuk mendapatkan daftar grup dengan ID nya maka gunakan endpoint baru yaitu `GET /groups`.

```bash
curl 'https://api.kirimwa.id/v1/groups' \
  -H 'Authorization: Bearer API_TOKEN'
```

Contoh output dari perintah diatas.

```json
{
  "message": "Contacts has been synced.",
  "data": [
    {
      "id": "6281234567890-1628639680",
      "name": "Grup Keluarga Besar"
    },
    {
      "id": "6281234567890-1628639662",
      "name": "Developer Tim"
    }
  ]
}
```

Saya tidak sabar untuk mendengar feedback anda tentang fitur baru ini. Silahkan langsung dicoba 😎 .

Bagi yang belum memiliki API Token silahkan mendaftar di [https://developer.kirimwa.id](https://developer.kirimwa.id).

**Comment by ulilamri419** [comment]
Dear Mas Rio,

Wah, senang sekali mendengar kabar ini,.. namun
Saya gagal mendapatkan group list nya. error nya Message 404 not found.

<img width="879" alt="Error cek-group" src="https://user-images.githubusercontent.com/70904189/129350881-eebcb682-f358-42ca-b31f-b75a983554b3.png">

Bisa di cek kembali.

terima kasih.
BR
Amri

**Reply by rioastamal** [reply]
Halo mas @ulilamri419, apa bisa dicoba lagi dan bagaimana hasilnya?

**Reply by ulilamri419** [reply]
Masih Sama error nya. "message":"404 not found."

**Reply by rioastamal** [reply]
Mas @ulilamri419, 

Endpoint yang benar adalah `GET /v1/groups`. Opsi `device_id` disuplai lewat query string bukan body. Jika anda menggunakan `-d` maka cURL berasumsi anda ingin menggunakan `POST /v1/groups`. Cek lagi dokumentasi.

**Reply by ulilamri419** [reply]
Mas @rioastamal 

Alhamdulillah, sudah bisa, ternyata setelah saya baca dokumentasi, mungkin perlu di perbaiki mas di dokumentasi hanya menyebutkan CURL spt ini:
![image](https://user-images.githubusercontent.com/70904189/129432629-3bfed3a4-4901-433a-a2a5-c44e344f08ea.png)

lebih lengkapnya sbb:
**curl 'https://api.kirimwa.id/v1/groups?device_id=DEVICE_ID' \
  -H 'Authorization: Bearer API_TOKEN'**

Anyway, Good job mas Rio.. really appreciate.

Terima kasih,
Salam
Amri

**Reply by rioastamal** [reply]
Dokumentasi sudah diperbaiki.
